app/src/main/java/com/sopt/androidstudy/data/datasources/FriendDataSoures.kt

```diff
@@ -0,0 +1,123 @@
+package com.sopt.androidstudy.data.datasources
+
+import android.util.Patterns
+import androidx.lifecycle.LiveData
+import com.sopt.androidstudy.data.model.db.Friend
+import com.sopt.androidstudy.data.model.db.FriendDAO
+import com.sopt.androidstudy.data.model.types.MBTI
+import com.sopt.androidstudy.data.model.types.MBTIFeatures
+import javax.inject.Inject
+import javax.inject.Singleton
+
+class FriendDataSources @Inject constructor(private val dao: FriendDAO) {
+
+    suspend fun insert(friend: Friend): Boolean =
+        if (Patterns.EMAIL_ADDRESS.matcher(friend.email).matches()) {
+            dao.insertFriend(friend)
+            true
+        } else false
+
+    suspend fun update(friend: Friend) {
+        dao.updateFriend(friend)
+    }
+
+    suspend fun delete(friend: Friend) {
+        dao.deleteFriend(friend)
+    }
+
+    suspend fun deleteAll() {
+        dao.deleteAll()
+    }
+
+    fun getAllFriends(): LiveData<List<Friend>> = dao.getAllFriends()
+
+    fun getMBTIFeatures(mbti: MBTI): List<MBTIFeatures>? {
+
+        return when (mbti) {
+            MBTI.ENFJ -> listOf<MBTIFeatures>(
+                MBTIFeatures.ENFJ1,
+                MBTIFeatures.ENFJ2,
+                MBTIFeatures.ENFJ3
+            )
```

Comment:
```suggestion
            MBTI.ENFJ -> listOf(MBTIFeatures.ENFJ1, MBTIFeatures.ENFJ2, MBTIFeatures.ENFJ3)
```

코틀린에서는 타입추론 기능이 있기때문에 위와같이 데이터타입 생략이 가능합니다~!

Reply:
더 깔끔하네요!! 좋습니다!